tests/rpc/spec/drawing_spec.js

```diff
@@ -0,0 +1,93 @@
+describe('Drawing', function(){
+
+    function handleEvent(name, handler) {
+        channel.handleEvent(name, handler);
+            handlersToClean.push({
+            name: name,
+            handler: handler
+        });
+    };
+
+    beforeEach(function(done) {
+        channel.onReady(function() {
+            // Channel is now ready and listening.
+            channel.resetState(function() {
+                // Reset map and event counter.
+                eventCounter = 0;
+                // Delay between tests
+                setTimeout(function() {
+                    done();
+                }, 0);
+            });
+        });
+    });
+
+    afterEach(function() {
+        // Spy callback.
+        expect(eventCounter).toEqual(1, "Event count does not match");
+        // Reset event handlers.
+        while (handlersToClean.length) {
+            var item = handlersToClean.shift();
+            channel.unregisterEventHandler(item.name, item.handler);
+        };
+    });
+
+    describe('DrawTools', function() {
+
+        describe('Start Drawing', function() {
+
+            all("Drawing requests",[
+               'Point',
+               'LineString',
+               'Polygon',
+               'Circle',
+               'Box',
+               'Square'
+               ],
+                function(drawRequests, done) {
+                    channel.postRequest('DrawTools.StartDrawingRequest', ['my functionality id', drawRequests]);
+                    channel.log("StartDrawingRequest:", drawRequests)
+                    eventCounter++;

```

Comment:
I think this variable should be scoped from global to test scope (introduced / test file). In here it's not counting events so the naming is very misleading.

Reply:
Agreed, it is used to spy rpc callback in general. It was originally used to only spy event callbacks. Name is misleading.